pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -270,3 +271,84 @@ func (r *Revision) GetLastPinned() (time.Time, error) {
 func (rs *RevisionStatus) duck() *duckv1beta1.Status {
 	return &rs.Status
 }
+
+// PropagateDeploymentStatus takes the Deployment status and applies its values
+// to the Revision status.
+func (rs *RevisionStatus) PropagateDeploymentStatus(original *appsv1.DeploymentStatus) {
+	ds := transformDeploymentStatus(original)
+	cond := ds.getCondition(deploymentConditionReady)
+	if cond == nil {
+		return
+	}
+
+	switch cond.Status {
+	case corev1.ConditionUnknown:
+		revCondSet.Manage(rs).MarkUnknown(RevisionConditionResourcesAvailable, cond.Reason, cond.Message)
+	case corev1.ConditionTrue:
+		revCondSet.Manage(rs).MarkTrue(RevisionConditionResourcesAvailable)
+	case corev1.ConditionFalse:
+		revCondSet.Manage(rs).MarkFalse(RevisionConditionResourcesAvailable, cond.Reason, cond.Message)
+	}
+}
+
+var depCondSet = apis.NewLivingConditionSet(
+	deploymentConditionProgressing,
+	deploymentConditionReplicaSetReady,
+)
+
+const (
+	// deploymentConditionReady means the underlying deployment is ready.
+	deploymentConditionReady = apis.ConditionReady
+	// deploymentConditionReplicaSetready inverts the underlying deployment's
+	// ReplicaSetFailure condition.
+	deploymentConditionReplicaSetReady apis.ConditionType = "ReplicaSetReady"
+	// deploymentConditionProgressing reflects the underlying deployment's
+	// Progressing condition.
+	deploymentConditionProgressing apis.ConditionType = "Progressing"
+)
+
+type deploymentStatus struct {
+	duckv1beta1.Status `json:",inline"`
+}
+
+func (ds *deploymentStatus) getCondition(t apis.ConditionType) *apis.Condition {
```

Comment:
Use this? https://github.com/knative/pkg/blob/a6e24ef7e5b297969d6a87a45ac3ec6d7b3c5f0f/apis/duck/v1beta1/status_types.go#L92